server/src/test/java/org/candlepin/dto/rules/v1/PoolTranslatorTest.java

```diff
@@ -75,19 +79,15 @@ protected Pool initSourceObject() {
         derivedProduct.setAttribute(Product.Attributes.STACKING_ID, "2221");
         source.setDerivedProduct(derivedProduct);
 
-        ProvidedProduct providedProd = new ProvidedProduct();
-        providedProd.setProductId("provided-product-id-1");
-        providedProd.setProductName("provided-product-name-1");
-        Set<ProvidedProduct> providedProducts = new HashSet<>();
-        providedProducts.add(providedProd);
-        source.setProvidedProductDtos(providedProducts);
+        Product providedProd = new Product();
+        providedProd.setId("provided-product-id-1");
+        providedProd.setName("provided-product-name-1");
+        source.getProduct().addProvidedProduct(providedProd);
```

Comment:
Move this block up and add it directly to the product instead of going indirectly through pool.

Reply:
Corrected the adding provided product to product.